src/mailbox-state.c

```diff
@@ -726,8 +732,14 @@ int mailbox_state_set_flags(struct mailbox_view *view,
 
 		if (*atom == '\\') {
 			/* system flag */
-			if (mail_flag_parse(atom + 1) == 0)
+			if ((flag = mail_flag_parse(atom + 1)) == 0)
+				return -1;
+			if ((flag & MAIL_RECENT) != 0 && imap4rev2_enabled) {
+				i_error("Mailbox FLAGS response contains "
+					"\\Recent flag, while IMAP4REV2 is "
+					"enabled");
```

Comment:
Well, one more complaint: Shouldn't split strings in the middle of sentences, since it makes grepping for them more difficult. I guess splitting from `\\Recent` is okay, but this `enabled` word should be in the previous line.

Reply:
Yeah i thought forth and back but couldn't decide what would be best, fixed now